packages/keystone/lib/ListTypes/list.js

```diff
@@ -1483,11 +1484,11 @@ module.exports = class List {
       // NOTE: These could return a Boolean or a JSON object (if using the
       // declarative syntax)
       getAccess: () => ({
-        getCreate: () => context.getListAccessControlForUser(this.key, undefined, 'create'),
-        getRead: () => context.getListAccessControlForUser(this.key, undefined, 'read'),
-        getUpdate: () => context.getListAccessControlForUser(this.key, undefined, 'update'),
-        getDelete: () => context.getListAccessControlForUser(this.key, undefined, 'delete'),
-        getAuth: () => context.getAuthAccessControlForUser(this.key),
+        getCreate: () => context.getListAccessControlForUser(this.key, undefined, 'create', undefined, context),
```

Comment:
this is tricky part. In Future we may be using list meta queries to determine the access to a list and item, this way if you decide to not run this query for each item then the feature will still remain incomplete. Some users have sought out that they want to only allow deletion if the relationship item has some value. 

top level list access call may not be accurate.

Top level access check using gql should only be done for `create` access. 

should there be item specific meta when you use read query?
```json
{ 
  data: {
    allUsers: [
      {
         name: "user1",
         // other fields ....
        ___meta: {
           access: { update: false, delete: true },
         },
      },
    ]
}
      

```